lib/FrameManager.js

```diff
@@ -577,7 +577,7 @@ class Frame {
       if (!waitForVisible && !waitForHidden)
         return true;
       const style = window.getComputedStyle(node);
-      const isVisible = style && style.visibility !== 'hidden' && hasVisibleBoundingBox();
+      const isVisible = style && style.visibility !== 'hidden' && style.display !== 'none' && style.opacity !== '0' && hasVisibleBoundingBox();
```

Comment:
` style.display !== 'none'` should be covered by the `hasVisibleBoundingBox()` check. If the element (or any ancestor) is `display: none` then the bounding rect will be 0 width/height.